calendar/api/parent/ext-calendar-provider.js

```diff
@@ -408,6 +449,21 @@ this.calendar_provider = class extends ExtensionAPI {
               };
             },
           }).api(),
+
+          onSend: new EventManager({
+            context,
+            name: "calendar.provider.onSend",
+            register: fire => {
+              let listener = (event, calendar, content) => {
+                return fire.async(convertCalendar(context.extension, calendar), content);
```

Comment:
Rant (not your fault): Wow, EventManager is mighty ugly code, for a single function call. You're simply doing `convertCalendar(context.extension, calendar)`, and all the rest of these 15 lines is just for the even listener. That should be simpler.